extensions/labs/text.js

```diff
@@ -0,0 +1,629 @@
+              }
+            }
+          },
+          {
+            opcode: 'setColor',
+            blockType: Scratch.BlockType.COMMAND,
+            text: 'set text color to [COLOR]',
+            arguments: {
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR
+              }
+            }
+          },
+          {
+            opcode: 'setWidth',
+            blockType: Scratch.BlockType.COMMAND,
+            text: 'set width to [WIDTH] aligned [ALIGN]',
+            arguments: {
+              WIDTH: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '200'
+              },
+              ALIGN: {
+                type: Scratch.ArgumentType.STRING,
+                menu: 'align'
+              }
+            }
+          }
+        ],
+        menus: {
+          // These all need acceptReporters: false for parity with the Scratch Labs version.
+          animate: {
+            acceptReporters: false,
+            items: ['type', 'rainbow', 'zoom']
+          },
+          font: {
+            acceptReporters: false,
+            items: [
+              ...NATIVE_FONTS,
+              {
+                text: 'random font',
+                value: 'Random'
+              }
+            ]
+          },
+          align: {
+            acceptReporters: false,
+            items: [
+              'left',
+              'center',
+              'right'
+            ]
+          }
+        }
+      };
+    }
+
+    /**
+     * @param {VM.Target} target
+     * @returns {TextState}
+     */
+    _getState (target) {
+      const state = target[CUSTOM_STATE_KEY];
+      if (!state) {
+        /** @type {TextState} */
+        const newState = {
+          skin: createTextCostumeSkin()
+        };
+        target[CUSTOM_STATE_KEY] = newState;
+        return newState;
+      }
+      return state;
+    }
+
+    /**
+     * @param {VM.Target} target
+     * @returns {boolean}
+     */
+    _hasState (target) {
+      return !!target[CUSTOM_STATE_KEY];
+    }
+
+    _hideAllText () {
+      for (const target of vm.runtime.targets) {
+        if (this._hasState(target)) {
+          this._hideText(target, this._getState(target));
+        }
+      }
+    }
+
+    /**
+     * @param {VM.Target} target
+     * @param {TextState} state
+     */
+    _renderText (target, state) {
+      state.skin.cancelAnimation();
+      renderer.updateDrawableSkinId(target.drawableID, state.skin.id);
+    }
+
+    /**
+     * @param {VM.Target} target
+     * @param {TextState} state
+     */
+    _hideText (target, state) {
+      state.skin.cancelAnimation();
+      target.setCostume(target.currentCostume);
+    }
+
+    setText ({ TEXT }, util) {
+      const state = this._getState(util.target);
+      state.skin.setText(Scratch.Cast.toString(TEXT));
+      this._renderText(util.target, state);
+      // TODO: Scratch might force 1 frame delay here? we probably shouldn't copy that though
```

Comment:
Scratch Lab *does* force a frame delay when running this block, but I agree with the comment that it probably shouldn't be added (because it's kinda annoying).

https://github.com/TurboWarp/extensions/assets/68464103/d9d94b92-0183-48cf-8f42-74f7743eb7dd